pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -89,6 +89,11 @@
     private int messageExpiryCheckIntervalInMinutes = 5;
     // How long to delay rewinding cursor and dispatching messages when active consumer is changed
     private int activeConsumerFailoverDelayTimeMillis = 1000;
+    // How long to delete inactive subscriptions from last consuming
+    // When it is 0, inactive subscriptions are not deleted automatically
+    private long subscriptionExpirationTimeMinutes = 0;
+    // How frequently to proactively check and purge expired subscription
+    private long subscriptionExpiryCheckIntervalInMinutes = 5;
```

Comment:
This setting is not in `conf/broker.conf`.

Reply:
In any case, still won't help if the broker gets restarted. 

Since the "inactivity" information is not persisted, the new broker will have to restart the counting from 0. 

For example, if `subscriptionExpiration` is 1 week, we might check every 5minutes but we cannot drop it for a week after the cosnumer disconnected. 

If the broker gets restarted after 6 days, the new broker will start the countdown again and the subscription will be stay there for ~2 weeks, or more, since there's no hard guarantee that this will ever happen.

Reply:
@merlimat 
I think if the check interval is frequent enough (e.g. 5 min), the expiration is not depending on broker restarts since it is judged from `md.getPublishTime()`:
```
if (System.currentTimeMillis() - md.getPublishTime() > expirationTime) {
```

Reply:
Ok, but isn't the publish time independent from a subscription being stale? 

Eg: I might have a topic for which there were no publishes for 1 week, but the consumer was always connected. If the consumer disconnects for 5minutes, I don't think we should drop the subscription.

Reply:
@merlimat 
I see, Umm...
Is there good way to compute the "last consume time"?

Reply:
I think that more than "last consume time" it should be more like "timestamp of last consumer disconnecting from a particular subscription"